apps/indexer/src/api/services/coingecko/coingecko.service.ts

```diff
@@ -37,4 +40,42 @@ export class CoingeckoService {
       });
     }
   }
+
+  async getTokenPrice(
+    tokenId: CoingeckoTokenId,
+    tokenContractAddress: string,
+    targetCurrency: string,
+  ): Promise<number> {
+    try {
+      const assetPlatform = CoingeckoIdToAssetPlatformId[tokenId];
+      const response = await fetch(
+        `${this.coingeckoApiUrl}/simple/token_price/${assetPlatform}?contract_addresses=${tokenContractAddress}&vs_currencies=${targetCurrency}`,
+        {
+          headers: {
+            "x-cg-demo-api-key": this.coingeckoApiKey,
+          },
+        },
+      );
+
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+
+      const data = await response.json();
+      const price = CoingeckoTokenPriceDataSchema.parse(data);
+      const priceValue =
+        price[tokenContractAddress.toLowerCase()]?.[targetCurrency];
+
+      if (!priceValue) {
+        throw new Error("Unable to extract dataset");
+      }
```

Comment:
you could actually define the zod schema within this function, already validating the presence of the `tokenContractAddress` varaible